pkg/model/application.go

```diff
@@ -0,0 +1,17 @@
+package model
+
+import "errors"
+
+var (
+	ErrApplicationNotFound  = NotFoundError{errors.New("application not found")}
+	ErrApplicationNameEmpty = ValidationError{errors.New("application name can't be empty")}
+)
+
+func ValidateAppName(appName string) error {
+	// TODO(eh-am): check for invalid characters, min/max names?
+	// problem is that it needs to be in sync with existing data
+	if appName == "" {
+		return ErrApplicationNameEmpty
+	}
+	return nil
```

Comment:
There is a function: https://github.com/pyroscope-io/pyroscope/blob/508466ea3e2c440bb4e5ee6803f5cfd748d32b7b/pkg/flameql/flameql.go#L87-L97

However, we don't have the max size defined. I'd say, 1024 bytes should be sufficient (for label keys as well). Regarding backward compatibility – if we don't want to issue a major release, we should either accept the risk, or allow arbitrary length. I think the risk is acceptable